stdlib.ispc

```diff
@@ -3468,45 +3472,51 @@ __declspec(safe) static inline float log(float x_full) {
         int exponent;
 
         const int NaN_bits = 0x7fc00000;
-        const int Neg_Inf_bits = 0xFF800000;
```

Comment:
Where the log algorithm is coming from? Possibly we need to put a reference to the origin of the algorithm to make sure we are not breaking the license.

I know that algorithms are not copyrightable, only specific implementation is. But we'd like to understand the origin and give the credit.

Reply:
This is my own modification from the initial ISPC algorithm. I fixed the special values from the initial algorithm and then generated a new lower-order polynomial (with my a Python script optimization algorithm I written myself) since a it is a bit faster and the precision is enough thanks to the correction close to log(1). Finally, I added a correction based on the Taylor series at log(1). I just checked the functions/errors with the great Geogebra tool. 
Thus, there is not breaking any license :) .

Reply:
That's impressive to say the least!

Two things then:
1. Let's put implementation note before the algorithm to explain how you come up with it and what it precision characteristics are there. This information definitely belongs to the code, as we might find bugs in it, consider improvement, etc. And understanding the background is very important in this case.
2. Have you "shopped around" for the alternatives? I.e. have you compared it's precision and performance with SVML and Sleef alternatives? It would be good to understand where it stands compared other proved implementations. If needed we can borrow algorithms from both libraries, it just needs to be done properly (license, copyright, etc).

Reply:
Thank you.

1. There was already a comment in the code about the precision (in ULPs). I added information about how the method works.
2. I never used Sleef yet, but it seems interesting/promising. I'm going to give it a try, but I think this can be done later, independently of this PR.

Reply:
I finally performed a quick check/benchmark with Sleef (the one available on Ubuntu 24.04 LTS system packages) since is relatively simple in practice (as long as we use a standard C interface and not LLVM IR one which is apparently not documented).
The C interface might be slightly slower since there is a function call. However, the Sleef functions are not masked so adding a masking introduces slight overhead. In the end, the two might cancel out so the performance may be representative of what we can get with direct a IR interface (if any).
Sleef provide 2 groups of functions: ones with 1 ULP error (typically due to rounding), and ones with a 3.5 ULP error (let say 4). The former can be significantly slower (i.e. up to 10 time slower) than the later and I think 4 ULP is a good reference for a project like ISPC where people focus on performance (and also because the current ISPC math log/exp function have a 4 to 10 ULP error with the new PRs). Not to mention the performance of 4 ULP functions are more stable. See the benchmark for more information: https://sleef.org/nontrigsp.png and https://sleef.org/trigsp.png . 
I wrote a  "corrected" implementation (not pushed) so to get a <5 ULP error (instead of <10 ULP) in order not to mix apples and oranges when comparing this with Sleef. It simply use a 9-degreee polynomial instead of a (current) 8-degree one.
Regarding performance, on my machine (Zen2), I get (in clocks, for the entire 32-bit float range):
```
This PR implementation (default math):                   2.11e9
Corrected PR implementation:                   2.26e9
Sleef avx2-i32x8 with C interface and no masking:   2.73e9
```
Thus, in this case, Sleef is about 30% slower than the less-precise PR implementation and about 20% slower than the similarly-precise corrected PR implementation :) !
A part of the overhead might come from the C call but probably not all of it.
It shows that this PR implementation is pretty good since it is competitive with Sleef (which is itself competitive with the SVML based on the provided Sleef's benchmarks).

Overall, aside from this specific `log` implementation, IMHO, Sleef is pretty interesting for ISPC since it has the benefit to provide: 
- (pretty good) bounded-precision and support for special numbers (eg. NaN, Inf, subnormals, etc.)
- challenging performance
- certainly more tested functions than ISPC ones
- (fast) double-precision functions
- a flexible license certainly compatible with the ISPC one
- an alternative implementation (useful when an implementation has issues)